tracks/julia/exercises/pythagorean-triplet/mentoring.md

```diff
@@ -25,34 +55,51 @@ function pythagorean_triplets(n)
     return triplets
 end
 ```
-````
 
-Telling Julia the type of `triplets` allows it to store the values without indirection. Without an annotation some (all?) version of Julia will type `triplets` as `Vector{Any}`, which means a vector of pointers to heap-allocated objects, which is slower.
+Calculating c from a and b clearly eliminates a huge number of iterations and is the main reason for the speedup.
 
-A small improvement is to tighten up the loop bounds using some simple arithmetic and ceiling division (`cld`):
+A much more modest improvement is achieved by telling Julia the type of `triplets` so that it can store the values without indirection.
+Without an annotation some (all?) version of Julia will type `triplets` as `Vector{Any}`, which means a vector of pointers to heap-allocated objects, which is slower.
```

Comment:
All, I think. That said, the speed-up in this case is incredibly small for some reason:

`112.999 μs (58 allocations: 2.77 KiB)` vs `111.599 μs (6 allocations: 3.17 KiB)` for the cubic solution and n=100.

Reply:
I think the amount of time spent mallocing tuples will be negligible in all cases except maybe the linear one. I should probably change this to explain that this is nice to do anyway so that later code infers better. Might be a bit complicated?

Reply:
Maybe. I think it might be a bit complicated.

Maybe some general note like `It's best to avoid Any-type arrays whenever possible for performance reasons. You can find more details about why that is in the [Performance Tips](https://docs.julialang.org/en/v1/manual/performance-tips/index.html#man-performance-abstract-container)`?

Reply:
Even in the linear case the number of appends is tiny compared to the number of other operations, so I think it just doesn't matter. I'll mention that giving a more descriptive type annotation will help inference in functions that are consuming the output of this function.